pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -609,8 +609,11 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	controller.AddCephVersionLabelToDeployment(c.clusterInfo.CephVersion, deployment)
 	controller.AddCephVersionLabelToDeployment(c.clusterInfo.CephVersion, deployment)
 	k8sutil.SetOwnerRef(&deployment.ObjectMeta, &c.clusterInfo.OwnerRef)
+	p := cephv1.GetOSDPlacement(c.spec.Placement)
 	if !osdProps.onPVC() {
-		cephv1.GetOSDPlacement(c.spec.Placement).ApplyToPodSpec(&deployment.Spec.Template.Spec)
+		p.ApplyToPodSpec(&deployment.Spec.Template.Spec)
+	} else if p.NodeAffinity != nil || p.PodAffinity != nil || p.PodAntiAffinity != nil || p.Tolerations != nil || p.TopologySpreadConstraints != nil {
+		p.ApplyToPodSpec(&deployment.Spec.Template.Spec)
```

Comment:
In case of onPVC() is true, seems like we need to apply the placement in both cases. The `ApplyToPodSpec()` will do the right thing to not override conflicting settings if set in both places. 
```
osdProps.placement.ApplyToPodSpec(&deployment.Spec.Template.Spec)
p.ApplyToPodSpec(&deployment.Spec.Template.Spec)
```
Note that if the node affinity is specified in both places the `ApplyToPodSpec()` will merge it and not the other properties. From the original issue that doesn't look like a requirement, the fallback behavior was the important piece of it so we should be fine without merging the others.

Reply:
so according to my understanding from the above comment, we  can apply both 
```
osdProps.placement.ApplyToPodSpec(&deployment.Spec.Template.Spec)
p.ApplyToPodSpec(&deployment.Spec.Template.Spec)
```
and `ApplyToPodSpec()` will do all the job. 

And in case of `node affinity` in both places, the ApplyToPodSpec() will merge this but not other properties. But this is not important which respect issues.

@travisn  please correct if I'm wrong.